pkg/lib/verifier/rule_builder_util.go

```diff
@@ -17,112 +18,174 @@ import (
+//getPcrEventLogEqualsRules method will create PcrEventLogEqualsRule and return the rule
+//return nil if error occurs
+func getPcrEventLogEqualsRules(pcrs []types.PcrIndex, pcrLogData *types.PCRS, flavor *hvs.Flavor, marker common.FlavorPart) ([]rules.Rule, error) {
 
 	var results []rules.Rule
 
-	// iterate over the banks, collecting the values for each supplied index
-	// and create PcrEventLogEqualsExcluding rules (when present).
-	for bank, pcrMap := range flavor.Pcrs {
-		for _, index := range pcrs {
-			if expectedPcrEx, ok := pcrMap[index.String()]; ok {
+	if pcrLogData != nil {
+		expectedPcrEventLogEntry := types.EventLogEntryFC{
+			Pcr: types.PCR{
+				Index: pcrLogData.PCR.Index,
+				Bank:  pcrLogData.PCR.Bank,
+			},
+			TpmEvent: pcrLogData.EventlogEqual.Events,
+		}
 
-				expectedEventLogEntry := types.EventLogEntry{
-					PcrIndex:  index,
-					PcrBank:   types.SHAAlgorithm(bank),
-					EventLogs: expectedPcrEx.Event,
-				}
-				expectedPcr, _ := rules.FlavorPcr2ManifestPcr(&expectedPcrEx, types.SHAAlgorithm(bank), index)
-				rule, err := rules.NewPcrEventLogEqualsExcluding(&expectedEventLogEntry, expectedPcr, flavor.Meta.ID, marker)
-				if err != nil {
-					return nil, errors.Wrapf(err, "An error occurred creating a PcrEventLogEqualsExcluding rule for bank '%s', index '%s'", bank, index)
+		rule, err := rules.NewPcrEventLogEquals(nil, &expectedPcrEventLogEntry, uuid.Nil, marker)
+		if err != nil {
+			return nil, errors.Wrapf(err, "An error occurred creating a PcrEventLogEquals rule for bank '%s', index '%d'", pcrLogData.PCR.Bank, pcrLogData.PCR.Index)
+		}
+		results = append(results, rule)
+
+	} else {
+		// iterate over the banks, collecting the values for each supplied index
+		// and create PcrEventLogEquals rules (when present).
+		for bank, pcrMap := range flavor.Pcrs {
+			for _, index := range pcrs {
+				if expectedPcrEx, ok := pcrMap[index.String()]; ok {
+
+					expectedEventLogEntry := types.EventLogEntry{
+						PcrIndex:  index,
+						PcrBank:   types.SHAAlgorithm(bank),
+						EventLogs: expectedPcrEx.Event,
+					}
+
+					rule, err := rules.NewPcrEventLogEquals(&expectedEventLogEntry, nil, flavor.Meta.ID, marker)
+					if err != nil {
+						return nil, errors.Wrapf(err, "An error occurred creating a PcrEventLogEqualsExcluding rule for bank '%s', index '%s'", bank, index)
+					}
+
+					results = append(results, rule)
 				}
-
-				results = append(results, rule)
 			}
 		}
 	}
 
 	return results, nil
 }
 
-func getPcrEventLogEqualsRules(pcrs []types.PcrIndex, flavor *hvs.Flavor, marker common.FlavorPart) ([]rules.Rule, error) {
-
+//getPcrEventLogEqualsExcludingRules method will create PcrEventLogEqualsRule and return the rule
+//return nil if error occurs
+func getPcrEventLogEqualsExcludingRules(pcrs []types.PcrIndex, pcrLogData *types.PCRS, flavor *hvs.Flavor, marker common.FlavorPart) ([]rules.Rule, error) {
 	var results []rules.Rule
```

Comment:
Using variable names as 'rules' instead of 'results' is showing error because we are using package name called 'rules'. Hence it is conflicting. Can we use some other name like 'pcrRules'?